cli/ops/websocket.rs

```diff
@@ -0,0 +1,221 @@
+// Copyright 2018-2020 the Deno authors. All rights reserved. MIT license.
+use super::dispatch_json::{Deserialize, JsonOp, Value};
+use crate::op_error::OpError;
+use crate::state::State;
+use deno_core::ZeroCopyBuf;
+use deno_core::{CoreIsolate, CoreIsolateState};
+use futures::executor::block_on;
+use futures::future::{poll_fn, FutureExt};
+use futures::{SinkExt, StreamExt};
+use http::{Method, Request, Uri};
+use std::borrow::Cow;
+use tokio::net::TcpStream;
+use tokio_tungstenite::tungstenite::protocol::frame::coding::CloseCode;
+use tokio_tungstenite::tungstenite::protocol::CloseFrame;
+use tokio_tungstenite::tungstenite::{Error, Message};
+use tokio_tungstenite::{client_async_tls, MaybeTlsStream, WebSocketStream};
+
+pub fn init(i: &mut CoreIsolate, s: &State) {
+  i.register_op("op_ws_create", s.stateful_json_op2(op_ws_create));
+  i.register_op("op_ws_send", s.stateful_json_op2(op_ws_send));
+  i.register_op("op_ws_close", s.stateful_json_op2(op_ws_close));
+  i.register_op("op_ws_next_event", s.stateful_json_op2(op_ws_next_event));
+}
+
+#[derive(Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct CreateArgs {
+  url: String,
+  protocols: String,
+}
+
+pub fn op_ws_create(
+  isolate_state: &mut CoreIsolateState,
+  _state: &State,
+  args: Value,
+  _zero_copy: &mut [ZeroCopyBuf],
+) -> Result<JsonOp, OpError> {
+  let args: CreateArgs = serde_json::from_value(args)?;
+  let resource_table = isolate_state.resource_table.clone();
+  let future = async move {
+    let uri: Uri = args.url.parse().unwrap();
+    let request = Request::builder()
+      .method(Method::GET)
+      .uri(&uri)
+      .header("Sec-WebSocket-Protocol", args.protocols)
+      .body(())
+      .unwrap();
+    let domain = &uri.host().unwrap().to_string();
+    let port = &uri.port_u16().unwrap_or(match uri.scheme_str() {
+      Some("wss") => 443,
+      Some("ws") => 80,
+      _ => 0,
```

Comment:
was gonna use unreachable, but i get screamed at for that... a more elegant solution would be nice

Reply:
Seems to work now with unreachable...

Reply:
i guess sometime the lint rules must have changed